### ISSUE TITLE
Detect improper formatting of fluent calls

### ISSUE THREAD

**Issue by mkordas** [teamed/qulice#670]
According to [this article](http://www.yegor256.com/2014/10/23/paired-brackets-notation.html#fluent), fluent calls should be formatted like
```java
new Bar()
    .with(
        veryLongListOfArguments
    )
    .with(argument)
```
instead of
new Bar()
    .with(
        veryLongListOfArguments
    ).with(argument)
```

I think we can very easily detect the latter. We just need to create check prohibiting pattern `^\s+\).\w+\(` in all Java files. I've run this regex search over a few project and I haven't seen any false-positives.

**Comment by krzyk** [comment]
@mkordas the only problem I see is (aside from probable checkstyle indentation problem with the first call) is in Xembly:
```
        return new Directives()
            .add("address").set("127.0.0.1").up()
            .add("cont").set("11").up()
```
that would end up with:
```
        return new Directives()
            .add("address")
            .set("127.0.0.1")
            .up()
            .add("cont")
            .set("11")
            .up()
```

I'm not sure which one is cleaner.
But if we can disable it should be OK (peding next checkstyle release, at least it will have for RegexpSingleLine).

**Comment by krzyk** [comment]
@davvd valid bug

**Comment by mkordas** [comment]
@krzyk my pattern does trigger on your example

**Comment by davvd** [comment]
> @davvd valid bug

@krzyk I added `bug` tag to this ticket

**Comment by davvd** [comment]
@mkordas thank you for the ticket reported, I topped your account for 15 mins, transaction `56b31d730f8b322871000243`

**Comment by yegor256** [comment]
@mkordas to be honest, I like this more:

```
new Bar().with(
  veryLongListOfArguments
).with(argument)
```

Isn't it better than:

```
new Bar()
  .with(
    veryLongListOfArguments   
  )
  .with(argument)
```

What do you think?

**Comment by mkordas** [comment]
@yegor256 agreed, I also prefer the first version more. Can you please update your blog?

**Comment by yegor256** [comment]
@mkordas well, my formatting is NOT against the rule suggested in the blog. we still open the bracket on the same line where we close it. right?

**Comment by yegor256** [comment]
@mkordas I think the rule here should be: "put as much code as possible on one line, provided you don't violate the paired brackets rule"

**Comment by mkordas** [comment]
@yegor256 yes, it's not breaking paired brackets rule, but it's against example:
![image](https://cloud.githubusercontent.com/assets/5467276/13018916/b598dc22-d1cc-11e5-9b68-58429b525ad9.png)
I like NEW rule.

**Comment by yegor256** [comment]
@mkordas yes, in that example it's done this way, but it also can be done differently. it's not against the "paired brackets" rule. so, we need a new rule... maybe the one I suggested will work. we need to think about it a bit more and play with examples

**Comment by krzyk** [comment]
@davvd this is postponed

**Comment by davvd** [comment]
> @davvd this is postponed

@krzyk got it, "postponed" label here

**Comment by davvd** [comment]
> @davvd this is postponed

@krzyk all right, we'll find someone else for this task

**Comment by yegor256** [comment]
Fixed in 76b69b3f8 on `master`.

The thread was stuck on a disagreement: the reporter proposed a regex `^\s+\)\.\w+\(` that would flag `).with(argument)`, but @yegor256 explicitly [preferred](https://github.com/yegor256/qulice/issues/670#issuecomment-174966751) that compact form and sketched a different rule: *"put as much code as possible on one line, provided you don't violate the paired brackets rule."*

Implemented that sketched rule as a `RegexpSingleline` in `checks.xml`:

- **id**: `FluentCallMustNotStartLine`
- **pattern**: `^\s+\.\w+\(\s*$`
- **message**: *"A fluent call opening a multi-line block must be attached to the previous line"*

This forbids the exact tension the issue described: a `.method(` that opens a multi-line paren block is **not allowed to sit alone** on its own line — it must ride on the previous expression (either the prior chain stage or a `)` on the same line). So:

```java
// forbidden:
new Bar()
    .with(
        veryLongListOfArguments
    )

// allowed:
new Bar().with(
    veryLongListOfArguments
)
```

Single-line fluent stages (`.filter(x -> ...)`, `.map(...).collect(...)`, etc.) are untouched — those don't open multi-line blocks.

Six pre-existing spots in qulice itself matched the new rule and were refactored to attach the opener to the previous line: `DiamondOperatorCheck`, `ProhibitUnusedPrivateConstructorCheck`, `ProhibitPlainJunitAssertionsRule`, plus three tests. New `RegexpSinglelineCheck` tests (`rejectsFluentCallAloneOnLineOpeningMultiLineBlock` / `acceptsFluentCallAttachedToPreviousLine`) cover both directions with dedicated fixtures. Full `mvn install` passes (239 unit tests + 13 invoker ITs).
